src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1175,6 +1191,10 @@ private synchronized void updateServerState(){
     }
     
     public void shutdown() {
+        if (!isRunning()) {
+            LOG.debug("ZooKeeper server is not running, so not proceeding to shutdown!");
```

Comment:
minor: "... so not proceeding to shutdown!" -> "...  not proceeding with shutdown."